### PR TITLE
PUD-1241: Uplift dps gradle-spring-boot. and more!

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -13,6 +15,8 @@ spring:
     serialization:
       WRITE_DATES_AS_TIMESTAMPS: false
 
+  mvc.pathmatch.matching-strategy: ant_path_matcher
```
Comment (philsegal) [inline src/main/resources/application.yml:18]: Needed for springfox issues with SpringBoot 2.5+

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/RestConfiguration.kt [hunk 2]
```diff
@@ -53,7 +53,8 @@ class RestConfiguration {
 }
 
 object ManageRecallsApiJackson : ConfigurableJackson(
-  KotlinModule()
+  KotlinModule.Builder()
+    .build()
```
[the remark this answers is not in the document]
Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/RestConfiguration.kt:57]: Bit weird though to call `.build()` and then the `.as` and `.with` methods 🤷‍♂️

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/SpringFoxConfig.kt [hunk 3]
```diff
@@ -89,4 +102,47 @@ class SpringFoxConfig {
       }
     }
   }
+
+  // Fixes incompatibility with latest spring-boot: https://github.com/springfox/springfox/issues/3462
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/SpringFoxConfig.kt:106]: Maybe add "PUD-1231" to this comment ... as a ticket to do this swagger migration/uplift.

File: src/main/resources/application.yml (hunk 1 above)
Comment (AndyNewton) [reply src/main/resources/application.yml:16]: Comment with to-be-binned with "PUD-1231"?

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -83,6 +87,9 @@ management:
       enabled: true
     prometheus:
       enabled: true
+  info:
+    env:
+      enabled: true
```
[the remark this answers is not in the document]
Comment (AndyNewton) [reply src/main/resources/application.yml:90]: Slightly weird config but 🤷‍♂️

**Review by AndyNewton — APPROVED** [review]
Approved ... with a few QQs 🕉️

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/RestConfiguration.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/RestConfiguration.kt:57]: Builder seems to create the KotlinModule, but then we convert to some other type 🤷‍♂️

**Review by AndyNewton — APPROVED** [review]

**Pull request merged**